File: Framework/wearable_health/android/src/main/kotlin/se/lnu/thesis/wearable_health/enums/HealthDataType.kt
package se.lnu.thesis.wearable_health.enums

import androidx.health.connect.client.records.HeartRateRecord
import androidx.health.connect.client.records.Record
import androidx.health.connect.client.records.SkinTemperatureRecord
import kotlin.reflect.KClass

enum class HealthDataType(val value: String, val readDefinition: String) {
    HEART_RATE("heartRate", "android.permission.health.READ_HEART_RATE") {
        override fun toRecord(): KClass<out Record> {
            return HeartRateRecord::class
        }
    },
    SKIN_TEMPERATURE("skinTemperature", "android.permission.health.READ_SKIN_TEMPERATURE") {
        override fun toRecord(): KClass<out Record> {
           return SkinTemperatureRecord::class
        }
    },
    UNKNOWN("unknown", "unknown") {
        override fun toRecord(): KClass<out Record> {
            return Record::class
        }
    };

    companion object {
        fun getDataTypeByValue(value: String): HealthDataType {
            val healthDataType = HealthDataType.values().find { it.value == value }
            if (healthDataType == null) {
                return UNKNOWN
            }
            return healthDataType
        }

        fun getDataTypeByReadDefinition(readDefinition: String): HealthDataType {
            val healthDataType = HealthDataType.values().find { it.readDefinition == readDefinition}
            if (healthDataType == null) {
                return UNKNOWN
            }
            return healthDataType
        }
    }

    abstract fun toRecord(): KClass<out Record>
}

File: Framework/wearable_health/lib/provider/enums/health_data_type.dart
import 'dart:io';

import 'package:wearable_health/provider/enums/supported_platform.dart';

enum HealthDataType {
  heartRate(value: "heartRate", supportedPlatforms: [SupportedPlatform.android, SupportedPlatform.ios ]),
  skinTemperature(value: "skinTemperature", supportedPlatforms: [SupportedPlatform.android]),
  bodyTemperature(value: "bodyTemperature", supportedPlatforms: [SupportedPlatform.ios]),
  unknown(value: "unknown", supportedPlatforms: []);

  const HealthDataType({ required this.value, required this.supportedPlatforms });

  final String value;
  final List<SupportedPlatform> supportedPlatforms;

  factory HealthDataType.fromString(String input) {
    switch (input) {
      case "heartRate":
        return HealthDataType.heartRate;
      case "skinTemperature":
        return HealthDataType.skinTemperature;
      case "bodyTemperature":
        return HealthDataType.bodyTemperature;
      default:
        return HealthDataType.unknown;
    }
  }
}